giskard/testing/tests/data_quality.py

```diff
@@ -0,0 +1,299 @@
+
+    Returns:
+        dict: A dictionary with the completeness score for each column.
+    """
+    completeness_scores = {}
+    for column in dataset.df.columns:
+        column_data = dataset.df[column]
+        completeness_ratio = len(column_data.dropna()) / len(column_data)
+        completeness_scores[column] = completeness_ratio
+    return TestResult(messages=completeness_scores)
+
+@test(name="Data Range Test")
+def range_test(dataset: Dataset, column: str, min_value=None, max_value=None):
+    """
+    Test for checking if data in a column falls within a specified range.
+
+    Args:
+        dataset (Dataset): The dataset to test.
+        column (str): The column to check.
+        min_value (float, optional): The minimum valid value. Defaults to None.
+        max_value (float, optional): The maximum valid value. Defaults to None.
+
+    Returns:
+        TestResult: The result of the test.
+    """
+    column_data = dataset.df[column]
+    if min_value is not None and max_value is not None:
+        test_passed = all(min_value <= x <= max_value for x in column_data.dropna())
+    elif min_value is not None:
+        test_passed = all(min_value <= x for x in column_data.dropna())
+    elif max_value is not None:
+        test_passed = all(x <= max_value for x in column_data.dropna())
+    else:
+        raise ValueError("Neither min_value nor max_value were provided")
+    return TestResult(passed=test_passed)
+
+@test(name="Data Validity Test")
+def validity_test(dataset: Dataset, column: str, valid_values=None):
+    """
+    Test for checking if data in a column is in a set of valid values.
+
+    Args:
+        dataset (Dataset): The dataset to test.
+        column (str): The column to check.
+        valid_values (list, optional): A list of valid values. Defaults to None.
+
+    Returns:
+        TestResult: The result of the test.
+    """
+    if valid_values is None:
+        raise ValueError("valid_values must be provided")
+    column_data = dataset.df[column]
+    test_passed = all(x in valid_values for x in column_data.dropna())
+    return TestResult(passed=test_passed)
+
+@test(name="Data Correlation Test")
+def correlation_test(dataset: Dataset, column1: str = None, column2: str = None):
+    """
+    Test for analyzing correlations between two specific features.
+
+    Args:
+        dataset (Dataset): The dataset to test.
+        column1 (str, optional): The first column to check. Defaults to None.
+        column2 (str, optional): The second column to check. Defaults to None.
+
+    Returns:
+        TestResult: The result of the test,
+        containing the correlation between the two columns
+        (if provided) or the full correlation matrix.
+    """
+    correlation_matrix = dataset.df.corr()
+
+    if column1 is not None and column2 is not None:
+        correlation = dataset.df[[column1, column2]].corr().iloc[0, 1]
+        return TestResult(passed=True,
```

Comment:
We always return `True`, we might have a param `should_correlate` to do this check.

Example cases:
- We find a correlation between sexe feature and higher salary which indicate some bias in our dataset. 
- We find correlation between the house area and it's price which is totally expected.